### ISSUE TITLE
allow per environment ssh config

### ISSUE THREAD

**Issue by wouterdb** [inmanta/inmanta#541]
Currently, all agents started by the server share an SSH config, 

it would be better to have per environment keys, so that keys can not be used by other environments

**Comment by wouterdb** [comment]
related to #447

**Comment by bartv** [comment]
However, the use is slightly different. #477 is for accessing git repos and this is for accessing remote hosts.

**Comment by wouterdb** [comment]
correct, I've started work on both at the same time. The underlying mechanism is the similar.
(but somewhat tricky to test)
